File: server/user/gateways.ts
export interface UserRepository {
  exists(name: string): Promise<boolean>
  save(user: User): Promise<User>
  update(user: User): Promise<User>
  getByName(name: string): Promise<User>
}

export type User = {
  name: string,
  password: string,
  lastConnection: Date,
};

export interface IUserService {
  createUser(user: User): Promise<User>;
}

File: server/Services/Gateways.cs
using System.Threading.Tasks;
using plantio.Domain;

namespace plantio.Services {
    public interface UserRepository {
        Task<bool> ExistsWithName(string name);
        void Save(User user);
        void Update(User user);
        void Delete(User user);
        Task<User?> GetByName(string name);
    }

    public interface UserTokenizer {
        string Tokenize(User user);
    }
}